Fix advanced search breadcrumb

ThoWagen opened the pull request:
We realized that the breadcrumb in advanced searches does not include a link back to the search anymore.

Here is a quick fix, but I'm not sure if this is the best way to do it and if there are unwanted side effects. So I created this as a draft and as a starting point for a discussion.

Should we maybe instead call the query parameter for advanced searches `sid` instead of `edit`?

ThoWagen committed:
fix advanced search breadcrumb

demiankatz commented:
@ThoWagen, I don't object to changing the `edit` parameter of the advanced search screen to use `sid` instead for consistency. If it doesn't introduce too much complexity, it might be worth adding backward-compatible support for the edit parameter just in case (e.g. if referenced in a local custom template), but I think the likely impact is small.

demiankatz reviewed (changes requested):
One other suggestion: it's probably worth adding a Mink test to cover this. I think we already have a test about editing advanced searches, so it's likely just a matter of adding a new assertion to the existing test. (I can help find it if you're not sure where it is, though hopefully it's not too hard to find).

ThoWagen committed:
added mink test case

ThoWagen committed:
switch to sid

ThoWagen commented:
Thanks @demiankatz! I'm not very familiar with the search memory. But since you don't see any concerns here I just switched from 'edit' to 'sid'. Note that 'edit' is still supported and there is a comment including the 'legacy' keyword for now.

I also added a Mink test case.

demiankatz commented on themes/bootstrap5/templates/search/searchbox.phtml:
```diff
@@ -63,7 +63,7 @@
       <p class="adv_search_terms"><?=$this->transEsc('Your search terms')?>: "<strong><?=$this->escapeHtml($this->lookfor)?></strong>"</p>
       <p class="adv_search_links">
-        <a href="<?=$this->url($advSearch)?>?edit=<?=$this->escapeHtmlAttr($this->searchId)?>"><?=$this->transEsc('Edit this Advanced Search')?></a> |
+        <a href="<?=$this->url($advSearch)?>?sid=<?=$this->escapeHtmlAttr($this->searchId)?>"><?=$this->transEsc('Edit this Advanced Search')?></a> |
```
While we're editing this line anyway, maybe it would be cleaner to revise it to leverage the `'query'` option of the `url` helper for consistency with the other instances of generating edit links.

ThoWagen replied:
Good idea! Done

demiankatz commented on module/VuFind/src/VuFind/Search/Memory.php:
```diff
@@ -210,8 +210,10 @@ public function retrieveSearch()
         $sid = $this->request->getQuery('sid')
-            ?? $this->request->getPost('sid');
+            ?? $this->request->getPost('sid')
+            ?? $this->request->getQuery('edit');
```
I would be inclined not to support the `edit` parameter here. If the only consequence is that somebody using an outdated link will not see a complete breadcrumb trail, that's not really a serious problem, and it's unlikely to occur... but the possibility of negative side effects is non-zero (should there be an edit query parameter that means something else in different code), and thus probably worth avoiding.

ThoWagen replied:
Yes you are right. I reverted that change.

demiankatz reviewed (changes requested):
Thanks, @ThoWagen -- see below for a couple more thoughts, but I think this overall approach should work.

ThoWagen committed:
Merge branch 'dev' into pull-request/fix-advanced-search-breadcrumb

ThoWagen committed:
implemented suggestions

demiankatz reviewed (approved):
Thanks, @ThoWagen!
